pyaedt/modules/PostProcessor.py

```diff
@@ -1449,218 +1683,213 @@ def get_far_field_data(self, expression="GainTotal", setup_sweep_name='', domain
-        if not setup_sweep_name:
-            setup_sweep_name = self._parent.nominal_sweep
-        if self.post_solution_type not in report_type:
-            print("Solution not supported")
-            return False
-        if not plottype:
-            modal_data = report_type[self.post_solution_type]
-        else:
-            modal_data = plottype
+                if isinstance(a, list):
+                    variations.append(a)
+                else:
+                    variations.append([a])
+            i+=1
+        ycomponents = []
+        if plot_initial_response:
+            ycomponents.append("InitialImpulseResponse<{}.int_ami_rx>".format(ami_name))
+        if plot_intermediate_response:
+            ycomponents.append("IntermediateImpulseResponse<{}.int_ami_rx>".format(ami_name))
+        if plot_final_response:
+            ycomponents.append("FinalImpulseResponse<{}.int_ami_rx>".format(ami_name))
+        self.oreportsetup.CreateReport(plotname, "Standard", plot_type, setupname, ["NAME:Context", "SimValueContext:=",
+                                                                                    [55824, 0, 2, 0, False, False, -1,
+                                                                                     1, 0, 1, 1, "", 0, 0, "NUMLEVELS",
+                                                                                     False, "1", "PCID", False, "-1",
+                                                                                     "PID", False, "1", "SCID", False,
+                                                                                     "-1", "SID", False, "0"]],
+                                       variations, ["X Component:=", "__InitialTime", "Y Component:=", ycomponents])
+        return plotname
+
+
+    def create_ami_statistical_eye_plot(self, setupname, ami_name, variation_list_w_value, ami_plot_type="InitialEye",plotname=None):
+        """
+                Creates an AMI Statistical Eye Plot
+
+
+                Parameters
+                ----------
+                setupname: str
+                    Name of the setup
+                probe_id: str
+                    AMI Probe Name to use
+                variation_list_w_value: list
+                    list of variations with relative values
+                plot_type: str, Default ``"Rectangular Plot"``
+                    String containing the report type. Default is ``"Rectangular Plot"``. it can be ``"Data Table"``, ``"Rectangular Stacked Plot"`` and all other valid AEDT Report types
+                ami_plot_type: str, Default ``"InitialEye"``
+                    String containing the report AMI type. Default is ``"InitialEye"``. it can be ``"EyeAfterSource"``, ``"EyeAfterChannel"`` or ``"EyeAfterProbe"``
+                plotname: str, Optional
+                    Defines the plot name
+
+                Returns
+                -------
+                str
+                    plot name
+                """
```

Comment:
```suggestion
        """Create an AMI statistical eye plot.

        Parameters
        ----------
        setupname : str
            Name of the setup.
        probe_id : str
            AMI Probe Name to use
        variation_list_w_value : list
            List of variations with relative values.
        plot_type : str, optional
            String containing the report type. Default is ``"Rectangular Plot"``. It can be ``"Data Table"``, 
            ``"Rectangular Stacked Plot"``, or any other valid AEDT report types.
        ami_plot_type : str, optional
            String containing the report AMI type. Default is ``"InitialEye"``. It can be ``"EyeAfterSource"``,
            ``"EyeAfterChannel"`` or ``"EyeAfterProbe"``.
        plotname: str, optional
            The name of the plot.  Defaults to a unique name starting with ``"Plot"``.

        Returns
        -------
        str
           The name of the plot.
        """
```